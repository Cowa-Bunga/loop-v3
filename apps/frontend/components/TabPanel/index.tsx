import { Box } from '@mui/material';

export default function TabPanel({ children, value, index, ...other }) {
  return (
    <Box
      role="tabpanel"
      hidden={value !== index}
      id={`tabpanel-${index}`}
      aria-labelledby={`tab-${index}`}
      {...other}
    >
      {value === index && <Box sx={{ py: 3 }}>{children}</Box>}
    </Box>
  );
}
