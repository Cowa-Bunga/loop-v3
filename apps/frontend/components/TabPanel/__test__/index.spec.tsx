import React from 'react'
import { render } from '@testing-library/react'
import TabPanel from '../index'

describe('TabPanel', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <TabPanel value={1} index={1}>
        panel
      </TabPanel>
    )
    expect(baseElement).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })
})
