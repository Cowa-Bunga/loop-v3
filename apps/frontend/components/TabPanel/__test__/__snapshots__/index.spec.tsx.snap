// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel should render successfully 1`] = `
<body>
  <div>
    <div
      aria-labelledby="tab-1"
      class="MuiBox-root css-0"
      id="tabpanel-1"
      role="tabpanel"
    >
      <div
        class="MuiBox-root css-0"
      >
        panel
      </div>
    </div>
  </div>
</body>
`;
